test/linter/test-versions.js

```diff
@@ -111,33 +143,45 @@ function addedBeforeRemoved(statement) {
  * @returns {void}
  */
 function checkVersions(supportData, relPath, logger) {
+  const category = relPath.split('.')[0];
   const browsersToCheck = Object.keys(supportData);
+
   for (const browser of browsersToCheck) {
     if (validBrowserVersions[browser]) {
       /** @type {SimpleSupportStatement[]} */
-      const supportStatements = [];
-      if (Array.isArray(supportData[browser])) {
-        Array.prototype.push.apply(supportStatements, supportData[browser]);
-      } else {
-        supportStatements.push(supportData[browser]);
-      }
-
-      const validBrowserVersionsString = `true, false, null, ${validBrowserVersions[
-        browser
-      ].join(', ')}`;
-      const validBrowserVersionsTruthy = `true, ${validBrowserVersions[
-        browser
-      ].join(', ')}`;
+      const supportStatements = Array.isArray(supportData[browser])
+        ? supportData[browser]
+        : [supportData[browser]];
 
       let sawVersionAddedOnly = false;
 
       for (const statement of supportStatements) {
+        if (statement === undefined) {
+          if (blockList[category].includes(browser)) {
+            logger.error(
+              chalk`{red → {bold ${browser}} must be defined for {bold ${relPath}}}`,
+            );
+          }
+
+          continue;
+        }
         const statementKeys = Object.keys(statement);
 
         for (const property of ['version_added', 'version_removed']) {
-          if (!isValidVersion(browser, statement[property])) {
+          if (
```

Comment:
This block would profit from extracting a local variable `version`, which could also be used instead of `statement.version_removed`:

```
const version = statement[property];
```